core/src/main/java/com/linecorp/armeria/internal/common/metric/ExecutorServiceMetrics.java

```diff
@@ -0,0 +1,322 @@
+        requireNonNull(tags, "tags");
+
+        new ExecutorServiceMetrics(executor, executorServiceName, metricPrefix, tags).bindTo(registry);
+        return new TimedScheduledExecutorService(registry, executor, executorServiceName, metricPrefix, tags);
+    }
+
+    /**
+     * Record metrics on the use of a {@link ScheduledExecutorService}.
+     *
+     * @param registry the registry to bind metrics to.
+     * @param executor the scheduled executor to instrument.
+     * @param executorServiceName will be used to tag metrics with "name".
+     * @param metricPrefix the prefix of the metric name.
+     * @param tags tags to apply to all recorded metrics.
+     * @return the instrumented scheduled executor, proxied.
+     * @since 1.3.0
+     */
+    public static ScheduledExecutorService monitor(MeterRegistry registry, ScheduledExecutorService executor,
+                                                   String executorServiceName, String metricPrefix,
+                                                   Tag... tags) {
+        requireNonNull(tags, "tags");
+        return monitor(registry, executor, executorServiceName, metricPrefix, asList(tags));
+    }
+
+    /**
+     * Returns whether the specified {@link Executor} is monitored by {@link MeterRegistry}.
+     */
+    public static boolean isMonitoredExecutor(Executor executor) {
+        return executor instanceof TimedScheduledExecutorService ||
+               executor instanceof TimedExecutorService ||
+               executor instanceof TimedExecutor ||
+               executor instanceof io.micrometer.core.instrument.internal.TimedScheduledExecutorService ||
+               executor instanceof io.micrometer.core.instrument.internal.TimedExecutorService ||
+               executor instanceof io.micrometer.core.instrument.internal.TimedExecutor;
+    }
+
+    /**
+     * Every ScheduledThreadPoolExecutor created by {@link Executors} is wrapped. Also,
+     * {@link Executors#newSingleThreadExecutor()} wrap a regular {@link ThreadPoolExecutor}.
+     */
+    @Nullable
+    private static ThreadPoolExecutor unwrapThreadPoolExecutor(ExecutorService executor, Class<?> wrapper) {
+        try {
+            final Field e = wrapper.getDeclaredField("e");
+            e.setAccessible(true);
+            return (ThreadPoolExecutor) e.get(executor);
+        } catch (NoSuchFieldException | IllegalAccessException e) {
+            // Do nothing. We simply can't get to the underlying ThreadPoolExecutor.
+        }
+        return null;
+    }
+
+    @Nullable
+    private final ExecutorService executorService;
+    private final String metricPrefix;
+    private final Iterable<Tag> tags;
+
+    private ExecutorServiceMetrics(@Nullable ExecutorService executorService, String executorServiceName,
+                                   String metricPrefix, Iterable<Tag> tags) {
+        this.executorService = executorService;
+        this.metricPrefix = metricPrefix;
```

Comment:
Just wondering if we could check this is an empty string or not, so metric doesn't look `.completed`

Reply:
That sounds good. 👍

Reply:
Added for `executorServiceName` and `metricPrefix`